odl/solvers/functional/default_functionals.py

```diff
@@ -2101,6 +2102,87 @@ def __repr__(self):
                                          self.__norm.outernorm.exponent,
                                          self.__norm.pwisenorm.exponent)
 
+
+class MoreauEnvelope(Functional):
+    """Moreau envelope of a convex functional.
+
+    The Moraeu evelope is a way to smooth an arbitrary convex functional
+    such that its gradient can be computed given the proximal of the original
+    functional and has the same critical points as the functional.
+
+    It is also called the Moreau-Yosida regularization.
+
+    Note that the only computable property of the Moreau envelope is the
+    gradient, the functional itself cannot be evaluated efficiently.
+
+    See `Proximal Algorithms`_ for more information.
+
+    Notes
+    -----
+    The Moreau envelope of a convex functional
+    :math:`f : \mathcal{X} \\rightarrow \mathbb{R}` multiplied by a scalar
+    :math:`\\sigma` is defined by
+
+    .. math::
+        \mathrm{env}_{\\sigma  f}(x) =
+        \\inf_{y \\in \\mathcal{X}}
+        \\left\{ \\frac{1}{2 \\sigma} \| x - y \|_2^2 + f(y) \\right\}
+
+    The gradient of the envelope is given by
+
+    .. math::
+        [\\nabla \mathrm{env}_{\\sigma  f}](x) =
+        \\frac{1}{\\sigma} (x - \mathrm{prox}_{\\sigma  f}(x))
+
+    References
+    ----------
+    .. _Proximal Algorithms: \
+https://web.stanford.edu/~boyd/papers/pdf/prox_algs.pdf
+    """
+
+    def __init__(self, functional, sigma=1.0):
+        """Initialize an instance.
+
+        Parameters
+        ----------
+        functional : `Functional`
+            The functional ``f`` in the definition of the Moreau envelope,
+            the function to be smoothed.
+        sigma : positive float
+            The scalar ``sigma`` in the definition of the Moreau envelope.
+            Higher sigma give a strong smoothing while smaller values give
```

Comment:
Perhaps shorter: "Larger values mean stronger smoothing." The other direction is clear.